src/client_side.cc

```diff
@@ -3953,9 +3950,13 @@ ConnStateData::terminateAll(const Error &rawError, const LogTagsErrors &lte)
     // may be set during the pipeline cleanup above
     connLeftovers_ = false;
 
-    if (intputToConsume && !inBuf.isEmpty()) {
-        debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
-        inBuf.clear();
+    if (!inBuf.isEmpty()) {
+        if (intputToConsume) {
+            debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
+            inBuf.clear();
+        } else {
+            bareError.update(error ? error : ERR_STREAM_FAILURE); // XXX: bareLogTagsErrors
```

Comment:
Before this change, bareError update logic was fairly simple/clear[^1]: If there are no transactions in the pipeline, then this function does not give `error` to any transaction, so we should store `error` in bareError (that will be used later, when creating and logging unparsed transaction, if any).

[^1]: I am not implying that old logic was correct.

Now bareError update code is completely ignoring the presence of pipeline transactions (and their use of `error`) and entangles bareError update with complex logic of determining whether any unparsed transactions exist.

Can we simplify PR changes/code by updating bareError where it was updated before, but doing so _unconditionally_, leaving complex (and buggy!) code that determines the presence of unparsed transactions intact? If ConnStateData ends up using updated bareError, great! If updated bareError remains unused, it is not a big deal. IIRC, that data member exists "just in case it is needed for future transaction objects".

Reply:
Done.